src/components/tables/PastJobsTable.tsx

```diff
@@ -64,19 +67,26 @@ const PastJobsTable = () => {
   }, [data]);
 
   return (
-    <BaseTable
-      data={tableData}
-      emptyView="No Past Jobs 😭"
-      isLoading={isLoading}
-      isError={isError}
-      refetch={refetch}
-      tableStructure={tableStructure}
-      tableHeaderComponent={
-        <Heading size="sm" mb={1}>
-          Past Jobs
-        </Heading>
-      }
-    />
+    <>
+      <BaseTable
+        data={tableData}
+        emptyView="No Past Jobs 😭"
+        isLoading={isLoading}
+        isError={isError}
+        refetch={refetch}
+        tableStructure={tableStructure}
+        tableHeaderComponent={
+          <Heading size="sm" mb={1}>
+            Past Jobs
+          </Heading>
+        }
+      />
+      <Pagination
+        pages={data?.totalPages || 0}
```

Comment:
data?.totalPages is lost on each page change. 
You can persist the pageCount a in ref, then have a memo be the must uptodate version with a [data] dependency array which updates the ref on each change. Or with a state and effect.

Reply:
okay, let me implement this.